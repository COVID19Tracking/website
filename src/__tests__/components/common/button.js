import React from 'react'
import renderer from 'react-test-renderer'
import {
  ButtonAnchor,
  ButtonLink,
  Button,
} from '../../../components/common/button'

describe('Components : Common: Button: Link component button', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ButtonLink to="/somewhere">An internal link</ButtonLink>)
      .toJSON()
    expect(tree).toMatchSnapshot()
    const bigTree = renderer
      .create(
        <ButtonLink to="/somewhere" big>
          A big internal link
        </ButtonLink>,
      )
      .toJSON()
    expect(bigTree).toMatchSnapshot()
  })
})

describe('Components : Common: Button: Anchor element button', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ButtonAnchor href="/somewhere">An external link</ButtonAnchor>)
      .toJSON()
    expect(tree).toMatchSnapshot()
    const bigTree = renderer
      .create(
        <ButtonAnchor href="/somewhere" big>
          A big external link
        </ButtonAnchor>,
      )
      .toJSON()
    expect(bigTree).toMatchSnapshot()
  })
})

describe('Components : Common: Button: Button element', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Button href="/somewhere">A button</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
    const bigTree = renderer
      .create(
        <Button href="/somewhere" big>
          A button
        </Button>,
      )
      .toJSON()
    expect(bigTree).toMatchSnapshot()
    const submitTree = renderer
      .create(
        <Button href="/somewhere" type="submit">
          A button
        </Button>,
      )
      .toJSON()
    expect(submitTree).toMatchSnapshot()
  })
})
