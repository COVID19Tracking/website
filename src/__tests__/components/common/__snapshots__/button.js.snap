// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Common: Button: Anchor element button renders correctly 1`] = `
<a
  className="button"
  href="/somewhere"
>
  An external link
</a>
`;

exports[`Components : Common: Button: Anchor element button renders correctly 2`] = `
<a
  className="button button-big"
  href="/somewhere"
>
  A big external link
</a>
`;

exports[`Components : Common: Button: Button element renders correctly 1`] = `
<button
  className="button"
  type="button"
>
  A button
</button>
`;

exports[`Components : Common: Button: Button element renders correctly 2`] = `
<button
  className="button button-big"
  type="button"
>
  A button
</button>
`;

exports[`Components : Common: Button: Button element renders correctly 3`] = `
<button
  className="button"
  type="submit"
>
  A button
</button>
`;

exports[`Components : Common: Button: Link component button renders correctly 1`] = `
<a
  className="button"
  href="/somewhere"
>
  An internal link
</a>
`;

exports[`Components : Common: Button: Link component button renders correctly 2`] = `
<a
  className="button button-big"
  href="/somewhere"
>
  A big internal link
</a>
`;
